plugins/permission-backend/src/service/AllowAllPermissionPolicy.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BackstageIdentity } from '@backstage/plugin-auth-backend';
+import { AuthorizeResult } from '@backstage/plugin-permission-common';
+import {
+  PermissionPolicy,
+  PolicyAuthorizeRequest,
+  PolicyResult,
+} from '@backstage/plugin-permission-node';
+
+export class AllowAllPermissionPolicy implements PermissionPolicy {
```

Comment:
Nope - it’s not even exported outside the package. However I’m questioning whether it should exist at all - right now it’s only used in standaloneServer.ts, but it doesn’t really make sense to provide a way to run the permission backend stand-alone but not override the policy. Seems like we should instead make the policy a required parameter there just like in createRouter - I’ll take a look into that tomorrow :).

Reply:
Hmm - this is actually a bit of an interesting problem. In summary: `standaloneServer.ts` is generally called by `run.ts`, which is intended to be run from the command line. As such, it's not possible to pass in a constructed `PermissionPolicy` in this context. It looks like this constraint occurs in other Backstage plugins too - for example, the jenkins-backend [expects a jenkinsInfoProvider](https://github.com/backstage/backstage/blob/a7085710d5a70f27d3c5da2570c5c5f81c95107a/plugins/jenkins-backend/src/service/router.ts#L26), but doesn't provide a way to customize it in standaloneServer.ts, and in fact [provides one](https://github.com/backstage/backstage/blob/master/plugins/jenkins-backend/src/service/standaloneServer.ts#L37) that doesn't look like it will work. More generally, almost all `standaloneServer.ts` implementations construct a `SingleHostDiscovery` instance for service discovery. It seems like _particularly_ for standalone servers, `SingleHostDiscovery` is likely to be the wrong mechanism for service discovery.

@Rugvip, have you got any thoughts on how to proceed here? Has there been much discussion about how plugin authors should approach allowing integrators to customize backends when running them via standaloneServer.ts?

Reply:
I think it's safe to say that the standaloneServer stuff is a historical remnant and an under-served thing that we haven't thought closely about in a long time. All real-world actual consumption of the backend plugins probably happens through creating the router thing instead, and then standing up your own backend rather than using the provided one. The standaloneServer things are probably, at best, usable for local dev (when you want to start just the single backend plugin while coding, rather than an example backend).

Reply:
When we look more closely at settling on a better contract for backend development in general, we might revisit this and see if we can make it a better-served use case. But for now, you could skip the file in its entirety if you like. The only outcome is that users who develop your plugin won't be able to do `yarn start` in your plugin dir.

Reply:
Nice, will remove the file for now, which means we can also remove the AllowAllPermissionPolicy :tada:.

Reply:
Done in 8f83584706